src/node/time.rs

```diff
@@ -1,82 +1,223 @@
 use anyhow::anyhow;
-use std::sync::{Arc, RwLock};
+use std::collections::VecDeque;
+use std::sync::{Arc, RwLock, RwLockReadGuard, RwLockWriteGuard};
+
+/// Shared readable view on time.
+pub trait TimeRead {
+    /// Returns the last timestamp (in seconds) that has already been used.
+    fn last_timestamp(&self) -> u64;
+
+    /// Peek at what the next call to `next_timestamp` will return.
+    fn peek_next_timestamp(&self) -> u64;
+}
+
+/// Writeable view on time management. The owner of this view should be able to treat it as
+/// exclusive access to the underlying clock.
+pub trait TimeExclusive: TimeRead {
```

Comment:
It's considered idiomatic to name traits as verbs (`Clone`, `Copy`, `Read`, `Write`) unless they represent an _interface_ that fully describes the object (`Future`, `Connection`), so probably the better names would be something like `ReadTime` and `AdvanceTime`.